subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java

```diff
@@ -393,6 +393,10 @@ private boolean dependencyMatchesProject(ModuleDependency dependency) {
     }
 
     private boolean isVersionMavenIncompatible(String version) {
```

Comment:
🧐  The meaning of this method changed, I think you should rename it `isMavenIncompatibleVersionInUse`.